lnwallet/contextsigner/internalsigner/interface.go

```diff
@@ -0,0 +1,133 @@
+package internalsigner
+
+import (
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/channeldb"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/lnwallet/chanfunding"
+)
+
+type internalSigner struct {
+	signer  input.Signer
+	keyRing keychain.KeyRing
+}
+
+func NewInternalSigner(
+	signer input.Signer,
+	keyRing keychain.KeyRing,
+) lnwallet.ChannelContextSigner {
+	return &internalSigner{
+		signer:  signer,
+		keyRing: keyRing,
+	}
+}
+
+func (is *internalSigner) NewChannel(
+	peerNode *btcec.PublicKey,
+	pendingChanID [32]byte,
+) error {
+	// The internal signer doesn't need this call, other
+	// ChannelContextSigner implementations do.
+	return nil
+}
+
+func (is *internalSigner) GetChannelBasepoints(
+	peerNode *btcec.PublicKey,
+	pendingChanID [32]byte,
+) (*lnwallet.ChannelBasepoints, error) {
+	var err error
+	var bps lnwallet.ChannelBasepoints
+	bps.MultiSigKey, err = is.keyRing.DeriveNextKey(
```

Comment:
Probably best to merge this into `NewChannel`, because `DeriveNextKey` is a mutating call, and must be called exactly once

Reply:
Also, lnd doesn't really have two different spots in the code for the two calls.  There's no reason to reflect the underlying lightning-signer API to the higher levels of lnd if lnd is structured differently.

Reply:
Done